modules/cli/src/integrationTest/java/org/apache/ignite/cli/ItClusterCommandTest.java

```diff
@@ -0,0 +1,256 @@
+
+    private static final List<Node> NODES = List.of(FIRST_NODE, SECOND_NODE, THIRD_NODE, FOURTH_NODE);
+
+    private static final String NL = System.lineSeparator();
+
+    private static final Logger topologyLogger = Logger.getLogger("org.apache.ignite.network.scalecube.ScaleCubeTopologyService");
+
+    /** DI context. */
+    private ApplicationContext ctx;
+
+    /** stderr. */
+    private final ByteArrayOutputStream err = new ByteArrayOutputStream();
+
+    /** stdout. */
+    private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+    @BeforeEach
+    void setup(@WorkDirectory Path workDir, TestInfo testInfo) throws Exception {
+        AtomicBoolean allNodesAreInPhysicalTopology = new AtomicBoolean(false);
+
+        Handler physicalTopologyWaiter = physicalTopologyWaiter(allNodesAreInPhysicalTopology);
+        topologyLogger.addHandler(physicalTopologyWaiter);
+
+        try {
+            startClusterWithoutInit(workDir, testInfo);
+
+            waitTillAllNodesJoinPhysicalTopology(allNodesAreInPhysicalTopology);
+        } finally {
+            topologyLogger.removeHandler(physicalTopologyWaiter);
+        }
+
+        ctx = ApplicationContext.run(Environment.TEST);
+    }
+
+    private Handler physicalTopologyWaiter(AtomicBoolean physicalTopologyIsFull) {
+        return new HandlerAdapter() {
+            @Override
+            public synchronized void publish(LogRecord record) {
+                if (record.getMessage().contains("Topology snapshot [nodes=" + NODES.size() + "]")) {
+                    physicalTopologyIsFull.set(true);
+                }
+            }
+        };
+    }
+
+    private void startClusterWithoutInit(Path workDir, TestInfo testInfo) {
+        CompletableFuture<?>[] futures = NODES.stream()
+                .map(node -> runAsync(() -> startNodeWithoutInit(node, workDir, testInfo)))
+                .toArray(CompletableFuture[]::new);
+
+        CompletableFuture.allOf(futures).join();
+    }
+
+    private void waitTillAllNodesJoinPhysicalTopology(AtomicBoolean allNodesAreInPhysicalTopology) throws InterruptedException {
+        assertTrue(IgniteTestUtils.waitForCondition(allNodesAreInPhysicalTopology::get, 10_000));
+    }
+
+    /**
+     * Initiates node start and waits till it makes its REST endpoints available, but does NOT invoke init.
+     *
+     * @param node      node
+     * @param workDir   working directory
+     * @param testInfo  test info
+     * @throws InterruptedException if we get interrupted
+     */
+    private void startNodeWithoutInit(Node node, Path workDir, TestInfo testInfo) throws InterruptedException, IOException {
+        String nodeName = testNodeName(testInfo, node.nodeIndex);
+
+        String config = configJsonFor(node);
+
+        IgnitionManager.start(nodeName, config, workDir.resolve(nodeName));
+
+        assertTrue(IgniteTestUtils.waitForCondition(() -> isHttpServedAt(node.restHostPort()), 10_000));
```

Comment:
There's no need for this, I think. After `InitionManager.start` completes, the REST manager should already be started

Reply:
Removed waiting for REST endpoint to be available. Also, added javadocs to `IgnitionManager`, `Ignition`, `IgniteImpl` to highlight that the node is half-started when `start()` method returns.